### PR TITLE
Custom condition

### PR DESCRIPTION
**Pull request by karatheodory**
Dear Sir,

In this pull request I have added a key generation function which allows the users to implement custom keys, not only users' IPs.

Thanks!

### CODE REVIEW

File: lib/express-rate-limit.js [hunk 1]
```diff
@@ -62,11 +66,16 @@ function RateLimit(options) {
...
+    }
 
+    // Backward compatibility function
     function resetIp(ip) {
-        delete hits[ip];
+        return resetKey(ip);
     }
 
+    rateLimit.resetKey = resetKey;
     rateLimit.resetIp = resetIp;
```
Comment (nfriedly) [inline lib/express-rate-limit.js]: You could just set `ratelimit.resetIp = resetKey;` and then get rid of the `resetIp()` function all together.

**Comment by nfriedly** [comment]
Hey, nice work @karatheodory, this looks like a smart change.

I left one comment about a slight simplification, and then I'd also like to see a test that validates the functionality. (Maybe include a token in the querystring and a custom function that uses the token instead of the IP - check that that one token gets blocked after too many requests, and also check that a second token does not get blocked once just because the first one is.

File: lib/express-rate-limit.js (hunk 1 above)
Comment (karatheodory) [inline lib/express-rate-limit.js]: That's right :smile: Thanks for your comment, I will fix it later today.

**Comment by karatheodory** [comment]
@nfriedly Hello! Can you please review the updated pull request?

**Comment by nfriedly** [comment]
Yep, it looks good. Nice work!

**Comment by nfriedly** [comment]
It should be released as v0.2.2 on npm automatically after Travis CI finishes it's thing.

**Pull request merged**